### PR TITLE
Source Tempo: migrate to lowcode + certify to beta

### LINKED ISSUE #18759: Source Tempo: migrate to low code

### LINKED ISSUE #18760: Source Tempo: Implement incremental syncs

### LINKED ISSUE #18761: Source Tempo: set up tests and docs

### PR DESCRIPTION
**Pull request by davydov-d**
## What
https://github.com/airbytehq/airbyte/issues/18682

## How
Migrate to low code, set up tests, edit docs etc

### CODE REVIEW

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        page_size: 50
+      page_size_option:
+          field_name: "limit"
+          inject_into: "request_parameter"
+      page_token_option:
+        inject_into: "path"
+      url_base: "*ref(definitions.requester.url_base)"
+  base_stream:
+    $options:
+      name: "accounts"
```
Comment (grubberr) [inline airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml:49]: Do we need it here?

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+      path: "accounts"
```
Comment (grubberr) [inline airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml:50]: Do we need it here?

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml [hunk 3]
```diff
@@ -0,0 +1,109 @@
+version: "0.1.0"
+
+definitions:
+  schema_loader:
+    file_path: "./source_tempo/schemas/{{ options['name'] }}.json"
```
Comment (grubberr) [inline airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml:5]: is it some default which can be removed from YAML ?

**Comment by grubberr** [comment]
is it ok?

```
$ python3 main.py discover --config secrets/config.json | jq .catalog.streams[].source_defined_primary_key
null
null
[
  [
    "tempoWorklogId"
  ]
]
null
```

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml (hunk 3 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml]: correct, removed it

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml (hunk 2 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml]: removed it

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml (hunk 1 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml]: removed

**Comment by davydov-d** [comment]
> @davydov-d is it ok?
> 
> ```
> $ python3 main.py discover --config secrets/config.json | jq .catalog.streams[].source_defined_primary_key
> null
> null
> [
>   [
>     "tempoWorklogId"
>   ]
> ]
> null
> ```

fixed it, thanks!

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    http_method: "GET"
+    error_handler:
+      type: CompositeErrorHandler
+      # ignore 403 error but retry default retriable http errors (429, 500 - 600)
+      error_handlers:
+        - type: DefaultErrorHandler
+          response_filters:
+            - http_codes: [403]
+              action: IGNORE
+        - type: DefaultErrorHandler
```
Comment (darynaishchenko) [inline airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml:22]: it is ok that `- type: DefaultErrorHandler` appears twice?

Comment (davydov-d) [reply airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml:22]: yep, that's done on purpose: first one processes only http status 400 responses, the second one - all others

**Review by darynaishchenko — APPROVED** [review]

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-tempo/acceptance-test-config.yml [hunk 5]
```diff
@@ -1,19 +1,40 @@
...
+        backward_compatibility_tests_config:
+          disable_for_version: "0.2.6"
   basic_read:
-    - config_path: "secrets/config.json"
-      configured_catalog_path: "integration_tests/configured_catalog.json"
+    tests:
+      - config_path: "secrets/config.json"
+        configured_catalog_path: "integration_tests/configured_catalog.json"
+        expect_records:
+          path: "integration_tests/expected_records.txt"
```
Comment (girarda) [inline airbyte-integrations/connectors/source-tempo/acceptance-test-config.yml:24]: were the expected records produced before or after the migration?

File: airbyte-integrations/connectors/source-tempo/source_tempo/client.py [hunk 6]
```diff
@@ -1,65 +0,0 @@
...
-# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
-#
-
-
-from functools import partial
-from json import JSONDecodeError
-from typing import Mapping, Tuple
-
-import requests
-from airbyte_cdk.sources.deprecated.client import BaseClient
```
Comment (girarda) [inline airbyte-integrations/connectors/source-tempo/source_tempo/client.py:11]: deleting a bunch of deprecated code :)

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+    $options:
+      name: "customers"
+      path: "customers"
+  worklogs_stream:
+    $ref: "*ref(definitions.base_stream)"
+    stream_cursor_field: "startDate"
+    retriever:
+      $ref: "*ref(definitions.retriever)"
+      requester: "*ref(definitions.requester)"
+      stream_slicer:
```
Comment (girarda) [inline airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml:66]: just to confirm - this is new right? The existing connector does not support incremental syncs

**Review by girarda — APPROVED** [review]
approved assuming the expected_records were created from the original implementation

File: airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml (hunk 7 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-tempo/source_tempo/tempo.yaml:66]: that's right

File: airbyte-integrations/connectors/source-tempo/acceptance-test-config.yml (hunk 5 above)
Comment (davydov-d) [reply airbyte-integrations/connectors/source-tempo/acceptance-test-config.yml:24]: Not quite, but there's a reason for that. Originally I generated them before the migration but within this request I have also made a migration to a new API version which made the response a little bit different from the one we used to have. Lots of links changed their values (you could notice schemas changed as well), at least the API version which is present in almost every link (in almost every record there's a link pointing to itself; besides, there's a lot of nested objects containing links as well). A sanity check was made to be sure things work as expected - a number of records yielded by each stream remains the same

**Pull request merged**